haskellPackages.{repa,hip}: Apply build fixes

rvl opened the pull request:
This PR is for merging into #279413, cc @sternenseemann.

## Description of changes

One does not simply `import Graphics.Image.Processing`.

This fixes the build of the `hip` package under ghc94, ghc96, ghc98. Because `repa` is a dependency, that also gets a patch so that it builds.

## Things done

```
nix build --system x86_64-linux .#haskell.packages.ghc9{4,6,8}.{hip,repa-{io,algorithms}}
```

I haven't opened upstream issues/PRs yet.

rvl committed:
haskellPackages.repa: Apply build fixes for ghc 9.4

rvl committed:
haskell.packages.ghc96.hip: Add override to fix build

rvl committed:
haskell.packages.ghc98.hip: Fix build of hip and dependencies

maralorn commented:
Can we maybe do this via the "inherit trick" so that Nix does not silently override potential other repa overrides in the file?

rvl commented:
> Can we maybe do this via the "inherit trick"

Yes, good idea @maralorn. I have updated the patch.

rvl committed:
haskellPackages.repa: Remove from broken.yaml

rvl committed:
haskellPackages: regenerate hackage-packages.nix

rvl committed:
haskellPackages: regenerate transitive-broken.yaml

rvl committed:
haskellPackages: regenerate hackage-packages.nix

domenkozar commented:
ping

rvl commented:
pong :slightly_smiling_face:

rvl commented:
I quoted the wrong PR number in the description, that's why.

rvl committed:
haskell.packages.ghc94.hip: Fix build of dependencies

rvl committed:
haskell.packages.ghc98.Chart-diagrams: Remove obsolete jailbreaks

Some dependencies of Chart-diagrams have been updated and no longer
require Cabal jailbreaks.

Of the remaining jailbreaks, I have added links to GitHub issues/PRs,
where they exist.

sternenseemann commented:
Cherry-picked onto haskell-updates with some minor changes.
